src/hardware/sblaster.cpp

```diff
@@ -2180,13 +2180,61 @@ OplMode find_oplmode()
 		case SBT_PRO1: opl_mode = OplMode::DualOpl2; break;
 		case SBT_2:
 		case SBT_1: opl_mode = OplMode::Opl2; break;
-		case SBT_GB: opl_mode = OplMode::Cms; break;
+		case SBT_GB: opl_mode = OplMode::None; break;
 		case SBT_NONE: opl_mode = OplMode::None; break;
 		}
 	}
 	return opl_mode;
 }
 
+bool is_cms_enabled()
+{
+	const auto* sect = static_cast<Section_prop*>(control->GetSection("sblaster"));
+	assert(sect);
+
+	const auto sbtype      = find_sbtype();
+	const bool cms_enabled = [sect, sbtype]() {
+		// Backward compatibility with existing configurations
+		if (sect->Get_string("oplmode") == "cms") {
+			LOG_WARNING("%s: The 'cms' setting for 'oplmode' is deprecated; use 'cms = on' instead.",
+			            CardType());
+			return true;
+		} else {
+			const auto cms_str = sect->Get_string("cms");
+			const auto cms_enabled_opt = parse_bool_setting(cms_str);
+			if (cms_enabled_opt) {
+				return *cms_enabled_opt;
+			} else if (cms_str == "auto") {
+				return sbtype == SBT_1 || sbtype == SBT_GB;
+			}
+			return false;
+		}
+	}();
+
+	switch (sbtype) {
+	case SBT_2: // CMS is optional for SoundBlaster 1, 1.5 and 2
+	case SBT_1: return cms_enabled;
+	case SBT_GB:
+		if (!cms_enabled) {
+			LOG_WARNING("%s: The 'cms' setting is 'off', but is forced to 'auto'.",
```

Comment:
We have the `GB: ` prefix when this is logged, but this would be a bit clearer:

`...but is forced to 'auto' on the Game Blaster.`